common/arch/sdl/window.cpp

```diff
@@ -72,21 +92,19 @@ int window_close(window *wind)
 		return 0;
 	}
 
-	if (wind == FrontWindow)
-		FrontWindow = wind->prev;
-	if (wind == FirstWindow)
-		FirstWindow = wind->next;
-	if (wind->next)
-		wind->next->prev = wind->prev;
-	if (wind->prev)
-		wind->prev->next = wind->next;
+	if (type == window_alloc_type::separate)
+		wind->unlink();		// Remove from the window list
 
 	if ((prev = window_get_front()))
 		WINDOW_SEND_EVENT(prev, EVENT_WINDOW_ACTIVATED);
 
+	// Callback needs to recognise this is a NULL pointer! (And 'wind' will be a dangling pointer for window_alloc_type::subclass)
```

Comment:
I misunderstood the point of the comment.  I interpreted it as a warning that `wind` might be a dangling pointer at the start of `w_callback`, and so some callbacks would need to ignore it or at least avoid dereferencing it.  However, your intent was that `wind` was guaranteed valid when `w_callback` begins, but might, depending on the callback, be invalid when `w_callback` returns.